Do not perform further validation on nullable null data

scaytrase opened the pull request:
Fixes #12

Commit:
Do not perform further validation on nullable null data

scaytrase commented on phpunit.xml:
```diff
@@ -6,8 +6,7 @@
          convertWarningsToExceptions="true"
-         processIsolation="false"
-         stopOnFailure="true">
+         processIsolation="false">
```
It's hard to debug multiple tests with `stopOnFailure=true`. I can revert it if you against this change

lezhnev74 replied:
It's okay, let's leave it as you suggested.

scaytrase commented on tests/FromCommunity/Issue3Test.php:
```diff
@@ -5,20 +5,17 @@
 use function GuzzleHttp\Psr7\stream_for;
 
-class ScayTraseTest extends TestCase
+class Issue3Test extends TestCase
```
I'm pretty proud for having named tests, but these are spawning fast. I suggest rename them according to fixed issue

lezhnev74 replied:
Haha, yeah. Good point.

scaytrase commented on tests/FromCommunity/Issue12Test.php:
```diff
@@ -0,0 +1,79 @@
+    public function getNullableTypeExamples(): array
+    {
+        return [
+            'nullable null' => [null],
```
if fix is reverted this test should fail

lezhnev74 replied:
Yeah, that is good. I also add at least two tests for a feature (red and green) to confirm both ways.

lezhnev74 committed:
Refactor Validator

I let it validate 'nullable' keyword as usual, and then if value is a valid Null, we just stop validation.

lezhnev74 committed:
Update comments

lezhnev74 commented:
OK, looks good to me. Merging.
